docs/en/operations/system-tables/distributed_ddl_queue.md

```diff
@@ -0,0 +1,72 @@
+# system.distributed_ddl_queue {#system_tables-distributed_ddl_queue}
+
+Contains information about distributed ddl queries (ON CLUSTER queries) that were executed on a cluster.
+
+Columns:
+
+-   `entry`  ([String](../../sql-reference/data-types/string.md)) - Query id.
+-   `host_name`  ([String](../../sql-reference/data-types/string.md)) - Hostname.
+-   `host_address`  ([String](../../sql-reference/data-types/string.md)) - IP address that the Hostname resolves to.
+-   `port`  ([UInt16](../../sql-reference/data-types/int-uint.md)) - Host Port.
+-   `status`  ([Enum](../../sql-reference/data-types/enum.md)) - Stats of the query.
+-   `cluster`  ([String](../../sql-reference/data-types/string.md)) - Cluster name.
+-   `value`  ([String](../../sql-reference/data-types/string.md)) - Node value.
+-   `query_start_time` ([Date](../../sql-reference/data-types/date.md)) — Query start time.
+-   `query_finish_time` ([Date](../../sql-reference/data-types/date.md)) — Query finish time.
+-   `query_duration_ms` ([DateTime64](../../sql-reference/data-types/datetime64.md)) — Duration of query execution in milliseconds.
+-   `exception_code`  ([Enum](../../sql-reference/data-types/enum.md)) - Exception code from ZooKeeper.
+
+
+**Example**
+
+``` sql
+SELECT *
+FROM system.distributed_ddl_queue
+WHERE (cluster = 'test_cluster') AND (entry = 'query-0000000000')
+LIMIT 2
+FORMAT Vertical
+
+Query id: 856f0efd-07c9-4f1a-815f-a19112df5bb3
+
+Row 1:
+──────
+entry:             query-0000000000
+host_name:         clickhouse01
+host_address:      172.23.0.11
+port:              9000
+status:            finished
+cluster:           test_cluster
+value:            version: 1
+query: CREATE DATABASE test_db UUID '40ac7692-70d3-48a9-bc29-4ade18957f59' ON CLUSTER test_cluster
+hosts: ['clickhouse01:9000','clickhouse02:9000','clickhouse03:9000','clickhouse04:9000']
+initiator: clickhouse01:9000
+
+query_start_time:  2020-12-15 10:06:35
+query_finish_time: 2020-12-15 10:06:35
+query_duration_ms: 7
+exception_code:    ZOK
+
+Row 2:
+──────
+entry:             query-0000000000
+host_name:         clickhouse02
+host_address:      172.23.0.12
+port:              9000
+status:            finished
+cluster:           test_cluster
+value:            version: 1
```

Comment:
Why not to parse the value using `DDLLogEntry::parse(...)` (see DDLWorker.cpp) and not to retrieve query from it? Also I'm not sure if we should print hosts list here, because we have `host_name`, `host_address` and `port` columns to print it.

Reply:
Thanks for the pointers, I can look into it. I agree that printing hosts here looks confusing.

Reply:
> Why not to parse the value using DDLLogEntry::parse(...) (see DDLWorker.cpp) and not to retrieve query from it?

I did not get this part:
What do you mean by "not to retrieve query from it" ? Do you mean use DDLLogEntry::parse(...)  to only retrieve relevant stuff like `query`, `version` and `initiator`?

@tavplubix if you could explain, that would be helpful.

Reply:
I mean "why ... not to retrieve query from it". Yes, we should parse the content of `query-xxxxxx` node using `DDLLogEntry::parse(...)` to retrieve `query`, `initiator` and `hosts`. And we should print `query`, `initiator` and `hosts` in separate columns, not in a single `value` column with plain-text content of `query-xxxxxxx`. 

As for `hosts`, we already have `host_name`, `host_address` and `port` columns. Values in these columns must be consistent with the hosts list parsed from `query-xxxxxxx`

Reply:
Thanks for clarifying I have made these changes.